### PR TITLE
Ruveyda

### PR DESCRIPTION
**Pull request by Ruveydabal**
I am still working on User and PremiumUser, but I would appreciate it if you could check if the code that i made for the other classes are good or if there are things that I should change.

### CODE REVIEW

**Comment by adev3141** [comment]
> I am still working on User and PremiumUser, but I would appreciate it if you could check if the code that i made for the other classes are good or if there are things that I should change.

I'll check it out and leave comments here

**Comment by Ruveydabal** [comment]
Oke thank you

File: StageGIM/Student Management/Student Management/Student.cs [hunk 1]
```diff
@@ -21,7 +21,7 @@ internal class Student
 
 
         public void AddStudents()
-        {     
+        {     //It just shows the list below you can't add, remove or serch for a specific student. Need to fix that
```
Comment (adev3141) [inline StageGIM/Student Management/Student Management/Student.cs]: please fix it and push again

File: StageGIM/StageGIM/IUser.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace LibraryManagementSystem
+{
+    internal interface IUser
```
Comment (adev3141) [inline StageGIM/StageGIM/IUser.cs]: this is good practice

File: StageGIM/StageGIM/PremiumUser .cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+using LibraryManagementSystem;
+
+
+namespace LibraryManagementSystem
+{
+    public class PremiumUser(string name, string userId, string email, string borrowedBooks, string membershipLevel, Library library) : User(name, userId, email, borrowedBooks)
```
Comment (adev3141) [inline StageGIM/StageGIM/PremiumUser .cs]: how is the premium user different from a normal user?

File: StageGIM/StageGIM/PremiumUser .cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    {
+        public string MembershipLevel { get; } = membershipLevel;
```
Comment (adev3141) [inline StageGIM/StageGIM/PremiumUser .cs]: where is membership level defined?

File: StageGIM/StageGIM/User.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using LibraryManagementSystem;
+
+namespace LibraryManagementSystem
+{    
+    public class User(string name, string userId, string email, string borrowedBooks): Library
```
Comment (adev3141) [inline StageGIM/StageGIM/User.cs]: user doesnt have membership level, what differentiates a premium user from a normal user?

File: StageGIM/Student Management/Student Management/Student.cs [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+
+namespace StudentRecordManagementSystem
+{
+    internal class Student
+    {
+        public string? StudentID { get; set; }
```
Comment (adev3141) [inline StageGIM/Student Management/Student Management/Student.cs]: why are these field nullable?

File: StageGIM/Student.cs [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace StudentRecordManagementSystem
+{
+    internal class Student
+    {
+        public string? StudentID { get; set; }
```
Comment (adev3141) [inline StageGIM/Student.cs]: why is this nullable?

**Review by adev3141 — COMMENTED** [review]
some explanations required

**Pull request closed without merging**